central/sensor/service/service_impl.go

```diff
@@ -119,23 +127,24 @@ func (s *serviceImpl) Communicate(server central.SensorService_CommunicateServer
 			capabilities = append(capabilities, centralsensor.ScannerV4Supported)
 		}
 
-		preferences := s.manager.GetConnectionPreference(cluster.GetId())
+		preferences := s.manager.GetConnectionPreference(clusterID)
 
 		// Let's be polite and respond with a greeting from our side.
 		centralHello := &central.CentralHello{
-			ClusterId:        cluster.GetId(),
+			ClusterId:        clusterID,
 			ManagedCentral:   env.ManagedCentral.BooleanSetting(),
 			CentralId:        installInfo.GetId(),
 			Capabilities:     capabilities,
 			SendDeduperState: preferences.SendDeduperState,
 		}
 
 		if err := safe.RunE(func() error {
-			certBundle, err := clusters.IssueSecuredClusterCertificates(cluster, sensorHello.GetDeploymentIdentification().GetAppNamespace(), nil)
+			namespace := sensorHello.GetDeploymentIdentification().GetAppNamespace()
+			certificateSet, err := securedclustercertgen.IssueSecuredClusterCerts(namespace, clusterID)
```

Comment:
This originates from a choice I made some months ago.

We had two existing ways to provide certificates from Central -> Sensor:
1. https://github.com/stackrox/rox/pull/7518 - when admission controller was added, this was implemented in CentralHello so that Sensor can retrieve `admission-control-tls`.
2. the local scanner cert refresh mechanism

I dind't want to have yet a 3rd mechanism for cert refresh, and I chose to extend (2) for it, because:
- (1) only works with CentralHello, with (2) I can also schedule refreshes while Sensor is running
- (2) had support for writing the TLS certs as k8s secrets. (1) is just saving them in an `emptyDir` and it's the services responsibility to retrieve the certs via an API

To answer your questions:
* `clusters.IssueSecuredClusterCertificates` only returns certs for sensor, collector and admission-control
* `securedclustercertgen.IssueSecuredClusterCerts` returns all certs needed (incl scanner-*), and it's the same function used by the cert refresh API (I added it in https://github.com/stackrox/stackrox/pull/12740)
* I think that the previous function is still used to create the legacy init bundles, but I may be wrong
* I think old clients should be fine, but we need to test. I think in practice old clients don't use `centralHello.CertBundle` anymore at all, because I doubt we still have customers with init bundles that don't have `admission-control-tls`. Basically old clients will just store more certificates in that `emptyDir` that is not really used.

Reply:
Thank you @vlad for chiming in.

Reply:
> I think in practice old clients don't use `centralHello.CertBundle` anymore at all, because I doubt we still have customers with init bundles that don't have `admission-control-tls`.

Not not having it is one use case. A much more frequent is when secured cluster components are installed in a namespace other than `stackrox` (the only one supported by init bundles) - I think in that case this admission control cert fetching also kicks in 🤔

Reply:
> A much more frequent is when secured cluster components are installed in a namespace other than stackrox (the only one supported by init bundles) - I think in that case this admission control cert fetching also kicks in

@porridge That is correct and I wasn't aware of this. We should make sure that this still works then (using an init bundle with a namespace that is not stackrox, that is).